skrf/media/cpw.py

```diff
@@ -83,74 +232,244 @@ def __repr__(self) -> str:
+        if diel == 'djordjevicsvensson':
+            # compute the slope for a log frequency scale, tanD dependent.
+            k = log((f_high + 1j * f_epr_tand) / (f_low + 1j * f_epr_tand))
+            fd = log((f_high + 1j * f) / (f_low + 1j * f))
+            ep_d = -tand * ep_r  / imag(k)
+            # value for frequency above f_high
+            ep_inf = ep_r * (1. + tand * real(k) / imag(k))
+            # compute complex permitivity
+            ep_r_f = ep_inf + ep_d * fd
+            # get tand
+            tand_f = -imag(ep_r_f) / real(ep_r_f)
+        elif diel == 'frequencyinvariant':
+            ep_r_f =  ep_r - 1j * ep_r * tand
+            tand_f = tand
+        else:
+            raise ValueError('Unknown dielectric dispersion model')
+        
+        return ep_r_f, tand_f
+    
+    def analyse_quasi_static(self, ep_r: NumberLike, 
+                           w: NumberLike, s: NumberLike,
+                           h: NumberLike, t: NumberLike,
+                           has_metal_backside: bool):
+        """
+        This function calculates the quasi-static impedance of a coplanar
+        waveguide line, the value of the effective permittivity as per filling
+        factor and the effective width due to the finite conductor thickness
+        for the given coplanar waveguide line and substrate properties.
+        Model from [#]_ with air backside and [#]_ with metal backside.
+        The models are corrected to account for
+        strip thickness using a first-order approach described in [GGBB96]_.
+        ADS simulator report to use a custom correction based on [Cohn60]_.
+        This second method is not implemented in skrf.
+        
+        References
+        ----------
+        .. [GhNa84] G. Ghione and C. Naldi. "Analytical Formulas for Coplanar Lines
+           in Hybrid and Monolithic MICs", Electronics Letters,
+           Vol. 20, No. 4, February 16, 1984, pp. 179-181.
+        .. [GhNa83] G. Ghione and C. Naldi. "Parameters of Coplanar Waveguides with
+            Lower Common Planes", Electronics Letters,
+            Vol. 19, No. 18, September 1, 1983, pp. 734-735.
+        .. [Cohn60] S. B. Cohn, "Thickness Corrections for Capacitive obstacles and
+           Strip Conductors", IRE Trans. on Microwave Theory and Techniques,
+           Vol. MTT-8, November 1960, pp. 638-644.
+        .. [GGBB96] K. C. Gupta, R. Garg, I. J. Bahl, and P. Bhartia, Microstrip
+           Lines and Slotlines, 2nd ed.Artech House, Inc., 1996.
+            
+        Returns
+        -------
+        zl_eff : :class:`numpy.ndarray`
+        ep_reff : :class:`numpy.ndarray`
+        """
+        Z0 = sqrt(mu_0 / epsilon_0)
+        a = w
+        b = w + 2. * s
+        
+        # equation (3a) from [GhNa84] or (6) from [GhNa83]
+        k1 = a / b
+        kk1 = ellipk(k1)
+        kpk1 = ellipk(sqrt(1. - k1 * k1))
+        q1 = ellipa(k1)
+        
+        # backside is metal
+        if(has_metal_backside):
```

Comment:
Style: It's more normal in Python to have space between if and condition and not use extra parentheses and this is also the style used in the rest of the skrf. The same thing few lines down.

Reply:
indeed ! I think I did write some C# just before and got mixed in my mind.